### PR TITLE
Refactored Weapon to be MonoBehaviour

### PR DESCRIPTION
**Pull request by Sirkular**
Weapon should be a MonoBehaviour to allow for easy static access to the object.
If we wanted to get the Weapon object from a GameObject previously, we had to do either player.GetComponent<PlayerControls>()._weapon or enemy.GetComponent<EnemyBehaviour>()._weapon. With this, we can just do gameObject.GetComponent<Weapon>();

### CODE REVIEW

File: Assets/Scripts/Weapons/Weapon.cs [hunk 1]
```diff
@@ -3,8 +3,9 @@
 using UnityEngine;
 
 //To be attached to Player or Enemy Objects
-public class Weapon
+public class Weapon : MonoBehaviour
```
Comment (zhuandy42) [inline Assets/Scripts/Weapons/Weapon.cs:6]: 😄

**Review by zhuandy42 — APPROVED** [review]
embrace the monobehaviour

File: Assets/Scripts/Enemy/EnemyFactory.cs [hunk 2]
```diff
@@ -56,6 +56,9 @@ public GameObject CreateEnemy(string enemyName, Vector3 position)
         // TODO: change default vector to dynamically adjust height of enemy spawn so they don't spawn under the ground.
         enemyTransform.position = position + new Vector3(0, 1, 0);
         
+        Weapon w = newEnemyObject.AddComponent<Weapon>();
```
Comment (MaXeraph) [inline Assets/Scripts/Enemy/EnemyFactory.cs:59]: we really shouldnt do these 1 letter variable names

**Review by MaXeraph — APPROVED** [review]
Otherwise looks good, boss.

File: Assets/Scripts/Enemy/EnemyFactory.cs (hunk 2 above)
Comment (Sirkular) [reply Assets/Scripts/Enemy/EnemyFactory.cs]: Haha, I thought that to myself as I typed it out. Fixed.

**Pull request merged**